tests/test_forward.py

```diff
@@ -168,6 +168,164 @@ def test_retina_ghm_forward():
                 batch_results.append(result)
 
 
+def test_cascade_empty_forward():
```

Comment:
It may be better to named the func as `test_cascade_forward` since it tests both empty and non-empty gt.